Source/Wikitivity.Agent/WikitivityAgentHelper.cs

```diff
@@ -5,43 +5,156 @@
+		{
+			QueryResultSlim test = new QueryResultSlim();
+			List<int> CasesWithWikitivity = new List<int>();
+
+			QueryRequest omGetInitialJobQuery = new QueryRequest()
+			{
+				ObjectType = new ObjectTypeRef() { Guid = WikitivityRDOGuid },
+				Condition = "('Request Url' ISSET)",
+				IncludeIDWindow = false,
+				RelationalField = null, 
+				SampleParameters = null,
+				SearchProviderCondition = null,
+				Sorts = null,
+				Fields = new List<FieldRef>()
 				{
-					success = true;
+					new FieldRef() {Name = "Request Url"},
+					new FieldRef() {Name = "Request ID"},
+					new FieldRef() {Name = "Page Title"}
+
 				}
+			};
+			try
+			{
+				test = await proxy.QuerySlimAsync(workspaceID, omGetInitialJobQuery, 1, 10000);
+			}
+			catch (Exception e)
+			{
+				Console.WriteLine(e);
+				throw;
+			}
+
+			foreach (var singleWorkspace in test.Objects)
+			{
+				CasesWithWikitivity.Add(singleWorkspace.ArtifactID);
+			}
+			return CasesWithWikitivity;
+		}
+		
+		public static async Task<QueryResultSlim> FindAllWikiCases(IObjectManager proxy, int workspaceID)
+		{
+
+
+			QueryRequest omGetInitialJobQuery = new QueryRequest()
+			{
+				ObjectType = new ObjectTypeRef() { Guid = WikitivityRDOGuid },
+				Condition = "('Request Url' ISSET)",
+				IncludeIDWindow = false,
+				RelationalField = null, //name of relational field to expand query results to related objects
+				SampleParameters = null,
+				SearchProviderCondition = null,
+				Sorts = null,
+				Fields = new List<FieldRef>()
+				{
+					new FieldRef() {Name = "Request Url"},
+					new FieldRef() {Name = "Request ID"},
+					new FieldRef() {Name = "Page Title"}
+				}
+			};
+
+			try
+			{
+				QueryResultSlim queryResults = await proxy.QuerySlimAsync(workspaceID, omGetInitialJobQuery, 1, 10000);
+				return queryResults;
 			}
 			catch (Exception ex)
 			{
-				throw new Exception("An error an occurred when deleting completed requests", ex);
+				return new QueryResultSlim();
 			}
+		}
 
-			return success;
+		public static async Task<QueryResultSlim> FindAllRequestsbyID(IObjectManager proxy, int workspaceID, string requestGuid)
+		{
+
+			QueryRequest findAllRequestsbyIDQuery = new QueryRequest()
+			{
+				ObjectType = new ObjectTypeRef() { Guid = WikitivityRDOGuid },
+				Condition = "('Request ID' IN ['"+requestGuid+"'])",
+				IncludeIDWindow = false,
+				RelationalField = null, //name of relational field to expand query results to related objects
+				SampleParameters = null,
+				SearchProviderCondition = null,
+				Sorts = null,
+				Fields = new List<FieldRef>()
+				{
+					new FieldRef() {Name = "Request Url"},
+					new FieldRef() {Name = "Request ID"},
+					new FieldRef() {Name = "Page Title"},
+					new FieldRef() {Name = "Name"}
+				}
+			};
+
+			try
+			{
+				QueryResultSlim queryResults = await proxy.QuerySlimAsync(workspaceID, findAllRequestsbyIDQuery, 1, 10000);

```

Comment:
You should avoid magic numbers like this.  Make them constants at the top of the class